apps/meteor/ee/app/livechat-enterprise/server/business-hour/Helper.ts

```diff
@@ -17,34 +19,44 @@ const getAllAgentIdsWithoutDepartment = async (): Promise<string[]> => {
 			},
 			{ projection: { _id: 1 } },
 		).toArray()
-	).map((user: any) => user._id);
+	).map((user) => user._id);
+
 	return agentIdsWithoutDepartment;
 };
 
-const getAgentIdsToHandle = async (businessHour: Record<string, any>): Promise<string[]> => {
+const getAgentIdsToHandle = async (businessHour: Pick<ILivechatBusinessHour, '_id' | 'type'>): Promise<string[]> => {
 	if (businessHour.type === LivechatBusinessHourTypes.DEFAULT) {
 		return getAllAgentIdsWithoutDepartment();
 	}
 	const departmentIds = (
 		await LivechatDepartment.findEnabledByBusinessHourId(businessHour._id, {
 			projection: { _id: 1 },
 		}).toArray()
-	).map((dept: any) => dept._id);
+	).map((dept) => dept._id);
 	return (
 		await LivechatDepartmentAgents.findByDepartmentIds(departmentIds, {
 			projection: { agentId: 1 },
 		}).toArray()
-	).map((dept: any) => dept.agentId);
+	).map((dept) => dept.agentId);
 };
 
-export const openBusinessHour = async (businessHour: Record<string, any>): Promise<void> => {
-	const agentIds: string[] = await getAgentIdsToHandle(businessHour);
+export const openBusinessHour = async (businessHour: Pick<ILivechatBusinessHour, '_id' | 'type'>): Promise<void> => {
+	businessHourLogger.debug(`Opening business hour ${businessHour._id}`);
+	const agentIds = await getAgentIdsToHandle(businessHour);
+	businessHourLogger.debug(`Opening business hour ${businessHour._id} for ${agentIds.length} agents`, {
```

Comment:
```suggestion
	businessHourLogger.debug({ msg: 'Opening business hour', businessHour: businessHour._id, agents: agentIds.length })
```

Reply:
I understand your concern about not logging all the agentIds in the log but I still think this type of info is quite useful for debugging. So, how about we only log max 10 agentIds here? This won't flood the logs and it would also be effective for debugging